source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,317 @@
+
+For a full list of client options, see :ref:`node-connection-options`.
+
+.. _node-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``MongoClientOptions`` so
+that the Server can validate them before the client connects. The
+driver provides multiple TLS-related client options to reference your
+certificates. You can connect with the following option configurations:
+
+- Passing a ``SecureContext`` object to the ``secureContext`` option *(Recommended)*
+- Passing filepath strings to the ``tlsCAFile`` and
+  ``tlsCertificateKeyFile`` options
+- Passing ``Buffer`` objects that contain the contents of your
+  certificates to the ``ca``, ``cert``, and ``key`` options
+
+.. _node-tls-securecontext:
+
+Create a ``SecureContext`` Object to Connect
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend that you use the ``secureContext`` option to connect with
+the TLS protocol. By using a ``SecureContext`` object to store your
+certificates, you use the Node.js framework for TLS that is independent of the
+driver. When you use this option, you keep all TLS options in a
+reusable object. This is the most organized way to set up
+TLS for your application.
+
+To create a ``SecureContext`` object, you must first install the ``tls``
+package and use it in your application. The ``tls.createSecureContext()``
+method takes an options parameter that contains the contents of the
+certificates. The ``secureContext`` option in your
+``MongoClientOptions`` takes the ``SecureContext`` object as its value,
+allowing the client to reference the certificates for validation.
+
+The following code shows how to create a ``SecureContext`` object by
+calling the ``tls.createSecureContext()`` method and set the
+``secureContext`` client option:
+
+.. code-block:: js
+   :emphasize-lines: 2-6, 10
+   
+   // Create a SecureContext object
+   var sc = tls.createSecureContext({
+     ca: fs.readFileSync(`<path to CA certificate>`),
+     cert: fs.readFileSync(`<path to client certificate>`),
+     key: fs.readFileSync(`<path to client certificate>`),
+   });
+   
+   // Pass the SecureContext as a client option
+   const client = new MongoClient(uri, {
+     secureContext: sc,
+   });
+
+To learn more about the ``tls.createSecureContext()`` method and the
+``tls`` package, see the `Node.js TLS API documentation
+<https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions>`__.
+
+To see a runnable example using the ``secureContext`` client option, see
+the :ref:`SecureContext Example <node-securecontext-full-example>`.
+
+Reference Certificate Filepaths to Connect
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can reference the filepaths for your certificates as strings in your
+client options to connect with TLS.
+
+The following code shows how to specify the ``tlsCAFile`` and
+``tlsCertificateKeyFile`` client options with certificate filepaths: 
```

Comment:
S: You're not exactly _referencing_ the filepaths--more like explicitly including them

```suggestion
To include the filepaths of your certificates as strings in your client options object,
use the ``tlsCAFile`` and ``tlsCertificateKeyFile`` options:
```

Reply:
reworded, but i think each section should still have an introductory sentence before the code intro